src/prlib/prlib.cpp

```diff
@@ -166,67 +260,155 @@ PR_EXTERN void PrSetStage(int stage)
+{
+    ((PrModelObject*)model)->m_contourBlurAlpha = blurAlpha;
+    ((PrModelObject*)model)->m_contourBlurAlpha2 = blurAlpha2;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrSetTransactionBlendRatio);
+PR_EXTERN void PrSetTransactionBlendRatio(PR_MODELHANDLE model, float blendRatio)
+{
+    ((PrModelObject*)model)->m_transactionBlendRatio = blendRatio;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetContourBlurAlpha);
+PR_EXTERN float PrGetContourBlurAlpha(PR_MODELHANDLE model)
+{
+    return ((PrModelObject*)model)->m_contourBlurAlpha;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetContourBlurAlpha2);
+PR_EXTERN float PrGetContourBlurAlpha2(PR_MODELHANDLE model)
+{
+    return ((PrModelObject*)model)->m_contourBlurAlpha2;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetTransactionBlendRatio);
+PR_EXTERN float PrGetTransactionBlendRatio(PR_MODELHANDLE model)
+{
+    return ((PrModelObject*)model)->m_transactionBlendRatio;
+}
 
-PR_EXTERN void PrSetModelDisturbance(PrModelObject *model, float disturbance)
+PR_EXTERN void PrSetModelDisturbance(PR_MODELHANDLE model, float disturbance)
 {
-    model->m_disturbance = disturbance;
+    ((PrModelObject*)model)->m_disturbance = disturbance;
 }
 
-PR_EXTERN float PrGetModelDisturbance(PrModelObject *model)
+PR_EXTERN float PrGetModelDisturbance(PR_MODELHANDLE model)
 {
-    return model->m_disturbance;
+    return ((PrModelObject*)model)->m_disturbance;
 }
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetVertexNum);
+PR_EXTERN u_int PrGetVertexNum(PR_MODELHANDLE model)
+{
+    return ((PrModelObject*)model)->m_spmImage->m_vertexNum;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetModelName);
+PR_EXTERN char* PrGetModelName(PR_MODELHANDLE model)
+{
+    return ((PrModelObject*)model)->m_spmImage->m_name;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetAnimationName);
+PR_EXTERN char* PrGetAnimationName(PR_ANIMATIONHANDLE animation)
+{
+    return ((SpaFileHeader*)animation)->m_name;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetCameraName);
+PR_EXTERN char* PrGetCameraName(PR_CAMERAHANDLE camera)
+{
+    return ((SpcFileHeader*)camera)->m_name;
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrGetSceneName);
+PR_EXTERN char* PrGetSceneName(PR_SCENEHANDLE scene)
+{
+    return ((PrSceneObject*)scene)->m_name;
+}
 
-PR_EXTERN PrRENDERING_STATISTICS* PrGetRenderingStatistics()
+PR_EXTERN void* PrGetRenderingStatistics()
 {
     PrRenderStuff* renderStuff = &prRenderStuff;
     return &renderStuff->m_renderStatistics;
 }
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrSetModelVisibility);
+PR_EXTERN void PrSetModelVisibillity(PR_MODELHANDLE model, u_int nodeIndex, u_int visible)
+{
+    SpmNode* spmNode;
+
+    if (nodeIndex < ((PrModelObject*)model)->m_spmImage->m_nodeNum) {
+        spmNode = ((PrModelObject*)model)->m_spmImage->m_nodes[nodeIndex];
+        if (visible) {
+            spmNode->m_flags &= 0xfffdffff;
+            return;
+        }
+        spmNode->m_flags |= 0x20000;
```

Comment:
```suggestion
        if (visible)
            spmNode->m_flags &= ~0x20000;
        else
            spmNode->m_flags |= 0x20000;
```
Can you try if this matches?